src/ConfigDiscoveryTrait.php

```diff
@@ -0,0 +1,151 @@
+ */
+
+namespace ZF\DevelopmentMode;
+
+use RuntimeException;
+
+/**
+ * Shared functionality for the Disable/Enable commands.
+ */
+trait ConfigDiscoveryTrait
+{
+    /**
+     * @var null|array
+     */
+    private $applicationConfig;
+
+    /**
+     * @var string
+     */
+    private $applicationConfigPath = 'config/application.config.php';
+
+    /**
+     * @var string Base name for configuration cache.
+     */
+    private $configCacheBase = 'module-config-cache';
+
+    /**
+     * @var null|string
+     */
+    private $configCacheDir;
+
+    /**
+     * @var null|string
+     */
+    private $configCacheKey;
+
+    /**
+     * Removes the application configuration cache file, if present.
+     *
+     * @param string $configCacheFile
+     */
+    private function removeConfigCacheFile($configCacheFile)
+    {
+        if (! $configCacheFile || ! file_exists($configCacheFile)) {
+            return;
+        }
+
+        unlink($configCacheFile);
+    }
+
+    /**
+     * Retrieve the config cache file, if any.
+     *
+     * @return false|string
+     */
+    private function getConfigCacheFile()
+    {
+        $configCacheDir = $this->getConfigCacheDir();
+        $configCacheKey = $this->getConfigCacheKey();
+
+        if (empty($configCacheDir)) {
+            return false;
+        }
+
+        $path = sprintf('%s/%s.', $configCacheDir, $this->configCacheBase);
+
+        if (! empty($configCacheKey)) {
+            $path .= $configCacheKey . '.';
+        }
+
+        return $path . 'php';
+    }
+
+    /**
+     * Return the configured configuration cache directory, if any.
+     *
+     * @return null|string
+     */
+    public function getConfigCacheDir()
+    {
+        if ($this->configCacheDir) {
+            return $this->configCacheDir;
+        }
+
+        $config = $this->getApplicationConfig();
+        if (isset($config['module_listener_options']['cache_dir'])) {
+            $this->configCacheDir = $config['module_listener_options']['cache_dir'];
+        }
+
+        return $this->configCacheDir;
+    }
+
+    /**
+     * Return the configured configuration cache key, if any.
+     *
+     * @return null|string
+     */
+    private function getConfigCacheKey()
+    {
+        if ($this->configCacheKey) {
+            return $this->configCacheKey;
+        }
+
+        $config = $this->getApplicationConfig();
+        if (isset($config['module_listener_options']['config_cache_key'])) {
+            $this->configCacheKey = $config['module_listener_options']['config_cache_key'];
+        }
+
+        return $this->configCacheKey;
+    }
+
+    /**
+     * Return the application configuration.
+     *
+     * Raises an exception if retrieved configuration is not an array.
+     *
+     * @return array
+     * @throws RuntimeException if config/application.config.php does not
+     *     return an array
+     */
+    function getApplicationConfig()
+    {
+        if ($this->applicationConfig) {
```

Comment:
it's not enough (when in previous execution set `$this-> applicationConfig = []` then this condition will be not satisfied - so checks below will be executed for each call):
https://3v4l.org/M0K41